### PR TITLE
feat: Convert video element to 'standard' element

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

Renames the `video` element to a `standard` element, and adds the supported types to the relevant data transformer.

A standard element represents every element covered by https://github.com/guardian/flexible-model/blob/f9d30ad2bb19446a9226ab1bd8418b4aaa03d762/src/main/thrift/content.thrift#L696.
 
In practice, it also covers the legacy (non-content-atom) audio and video elements, which only use a subset of their flexible-model fields.

This PR makes one modification to the element fields – every text-based element field is removed if empty, to match the current element behaviour.

Per the description in flexible-model, the data requirements for this element also match the `table` element, but that has a diferent presentation, and so we'll add it in a separate PR.

## How to test

It's worth testing with a variety of elements of different types, from different periods, in our consuming CMS' CODE environment. I've included a sample of pieces from that env, ordered by creation date, to check.

This PR contains fixture tests for each element type per #213, which offer some assurance that our new element type will not affect existing elements.

<details>
<summary>Pieces with video elements (these should be unchanged)</summary>
617805ee8f08f18a166abde6,2021-10-26
61d4740d8f08f06c8872433b,2022-01-04
6013f63a8f08be3f4348fd83,2021-01-29
5b8959e3e4b05cd0bedf4a7b,2018-08-31
610969698f088b0b4e9fdcd8,2021-08-03
619682ab8f08664e0af85c89,2021-11-18
5fc617948f086ce9248f3761,2020-12-01
5ff6e97f8f08c7a368e6e4ae,2021-01-07
600800b28f0850df55867fa4,2021-01-20
60ee4bb28f08331a6b2864e4,2021-07-14
60ee4bb28f08331a6b2864e4,2021-07-14
601a859b8f0837302b86d97c,2021-02-03
60cb61488f081fcfa2d35b01,2021-06-17
60cb61488f081fcfa2d35b01,2021-06-17
611bc8008f085022754cbe31,2021-08-17
60420bfc8f08e95b70aead0c,2021-03-05
6059c7cf8f083158d979fecf,2021-03-23
606dade58f081a7273c4dea2,2021-04-07
607d57568f08b5689bedf1e1,2021-04-19
5ad6f5eae4b04e8efbbb7457,2018-04-18
</details>

Screenshot 2022-03-01 at 09 04 04

<details>
<summary>Pieces with audio elements</summary>
60cb61488f081fcfa2d35b01,2021-06-17
603f69ec8f08c4b040ad3bde,2021-03-03
5f9add148f08e8dfdcc50fab,2020-10-29
5c88cac9e4b078b9d50aa516,2019-03-13
5b55a695e4b099be0ba37258,2018-07-23
5a844a86e4b0c73a428405e5,2018-02-14
5a69e1abe4b00f225a1c7aeb,2018-01-25
5a69ccfee4b0473d0bc10221,2018-01-25
5a6863a8e4b08c0147181fb2,2018-01-24
5a5ce770e4b046e0befbeae7,2018-01-15
5a5c9a62e4b0fe5e06bfa569,2018-01-15
5a5a48ffe4b0fe5e06bf9923,2018-01-13
5a55ffaae4b043ead794c45a,2018-01-10
5a4d0257e4b05efedfbd6fe0,2018-01-03
5a4bbd0ce4b01254e1d211b9,2018-01-02
5a4379b1b7f7b10656af30d9,2017-12-27
5a3bb77a7264a8077e24e74a,2017-12-21
5a3a9ffab428d606a9d9c8f9,2017-12-20
5a3a6bdfc7507c076d1334f6,2017-12-20
5a389bb0bbf0e7073eb10b2a,2017-12-19
</details>

Screenshot 2022-03-01 at 08 56 30

<details>
<summary>Pieces with map elements</summary>
60cb61488f081fcfa2d35b01,2021-06-17
5ce68b988f08f34f7281b941,2019-05-23
59c173c2e4b0f6beafd39d88,2017-09-19
59bcde75e4b06eeee7d810f8,2017-09-16
59471af4e4b014a733d638a5,2017-06-19
58ba0f6ee4b09c1763ef5eda,2017-03-04
58b5da25e4b030b6f7c901d6,2017-02-28
58ab6b8be4b030b6f7c8d717,2017-02-20
587fbbd8e4b01065cf597f03,2017-01-18
57ff8362e4b01ffb8ffcb4a4,2016-10-13
57dff371e4b0ac5fd0e5dfbf,2016-09-19
577094aee4b0f43038109f93,2016-06-27
57638603e4b088b489cb2488,2016-06-17
57619448e4b096841ef399b9,2016-06-15
5760c5f4e4b005ab05d211aa,2016-06-15
5760b806e4b0505cd1bcbc7c,2016-06-15
56dd68a1e4b0a02aed206487,2016-03-07
568331dae4b0629150aaf589,2015-12-30
567426b7e4b0fcd39b5714e2,2015-12-18
560e9348e4b06bd936f9c754,2015-10-02
</details>

Screenshot 2022-03-01 at 09 02 03

<details>
<summary>Pieces with document elements</summary>
60cb61488f081fcfa2d35b01,2021-06-17
6037b00b8f08d29f4c26eb46,2021-02-25
5aa2a3fbe4b00a04356fb5ef,2018-03-09
5a9e7be0e4b0f55650b88a7c,2018-03-06
5a232f930470380699b5386e,2017-12-02
59f74a62a885b806b2a4e7f6,2017-10-30
5951a06ce4b0cdea6d8657e4,2017-06-27
59160506e4b0a8ea08b6d9f9,2017-05-12
575915e5e4b01a5ff948b084,2016-06-09
57556bb8e4b0be576f9d9b55,2016-06-06
573c780ee4b04a0378340cf3,2016-05-18
57220c38e4b02046f02ebdc4,2016-04-28
56fa2eabe4b060534dcd586d,2016-03-29
56f1786fe4b0d8d383385a86,2016-03-22
56b3af27e4b05b93c48cd27b,2016-02-04
566bf8cae4b0c8f491ad28f9,2015-12-12
56588e65e4b03bf401a0bc0d,2015-11-27
56583dede4b0dd35d29c32eb,2015-11-27
559e8119e4b038c9e1db1b25,2015-07-09
554b4494e4b0c7299d72c995,2015-05-07
</details>

(Unsure how useful that preview is! 🤔)

Screenshot 2022-03-01 at 09 01 54

Tested in local Composer with all of the above, including altering captions and mandatory flags for each.

## How can we measure success?

A map, audio, and document element that works with the data we currently have. A video element that continues to work as expected.

## Accessibility

- [ ] [Tested with screen reader](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#screen-readers)
- [x] [Navigable with keyboard](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#keyboard-navigation)
- [x] [Colour contrast passed](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#colour-contrast)
- [x] [The change doesn't use only colour to convey meaning](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#use-of-colour)
- [x] [Interactive elements show a focus ring when focused by keyboard](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-the-focus-ring)
- [x] [Semantic elements have been used where possible](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-semantic-html)

### CODE REVIEW

File: src/plugin/fieldViews/CustomFieldView.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export interface CustomFieldDescription<Data = unknown, Props = unknown>
   props: Props;
 }
 
-export const createCustomField = <Data, Props>(
+export const createCustomField = <Data, Props = unknown>(
```
Comment (SHession) [inline src/plugin/fieldViews/CustomFieldView.ts:15]: Why was this needed?

**Review by SHession — COMMENTED** [review]
These changes make sense. However, I have spotted an issue with the way these new elements are added, it looks like the HTML field is being escaped causing them to render incorrectly.

File: src/plugin/fieldViews/CustomFieldView.ts (hunk 1 above)
Comment (jonathonherbert) [reply src/plugin/fieldViews/CustomFieldView.ts:15]: Ah, yeah, good to document. At the moment, we must always pass a value for the `props` argument in `createCustomField – we cannot make it optional, or the type passed to the type parameter and the value passed to the function might disagree, giving us 

```
Type 'Props | undefined' is not assignable to type 'Props'.
  'Props' could be instantiated with an arbitrary type which could be unrelated to 'Props | undefined'.
```

It felt silly to write

```ts
createCustomField<Asset[], undefined>([], undefined)
```

So adding `unknown` here lets us write 

```ts
createCustomField<Asset[]>([], undefined)
```

But the ideal would be 

```ts
createCustomField<Asset[]>([])
```

**Comment by jonathonherbert** [comment]
@SHession thanks for spotting, yes, #218 changed things underneath this PR! That should now be fixed.

**Review by SHession — COMMENTED** [review]
The serialisation issue has now been resolved!

I wonder if we could improve the UI to make better use of the space available. For instance making the caption field multiline and perhaps also using the black space to display a link to the original URL? See screenshot: 

image

**Comment by jonathonherbert** [comment]
Both good suggestions. There's a question of how far we should go – some of these content types fit nicely in the space they're given, some of them (document) need a _lot_ of space to preview nicely. So if we rearrange things but keep the aspect ratio, we end up with something that makes, I think, better use of the space:

Screenshot 2022-03-15 at 15 50 15

The problem is that documents do need lots of space to appear even remotely usable:

Screenshot 2022-03-15 at 15 50 24
 
And the aspect ratio of other elements means that if we give the document the full width,  we give other elements (which have aspect ratios that look something like 5:4 or 16:9) too much space, and we end up in a position where they're either too large (because we show them in their entirety) or they're less comprehensible (because we cut them off).

Document elements are the main problem here, and I'm tempted to say, they're really quite rare – let's add a little preview (arguably not _very_ useful) to give an indication that they're the right thing, with a link to the original content (great idea).

This should give the user an indication that they're looking at the correct content, which they can then inspect directly if they need to confirm or modify?

Interested in your thoughts.

**Comment by jonathonherbert** [comment]
Discussed out of band with @SHession and we're leaning towards 1. – I've pushed a commit with a new layout, lmk what you think, Sam

**Review by SHession — APPROVED** [review]
The UI look good, they make better use of the space. Changes LGTM!

`nit`: The video element iframe and url point to different videos (noticed because of the new content link)
